src/tablecloth/column/api/column.clj

```diff
@@ -0,0 +1,40 @@
+(ns tablecloth.column.api.column
+  (:require [tech.v3.dataset.column :as col]
+            [tech.v3.datatype :as dtype]))
+
+(defn column
+  "Create a `column` from a vector or sequence. "
+  ([]
+   (col/new-column nil []))
+  ([data]
+   (column data {:name nil}))
```

Comment:
`tech.ml.dataset`  makes heavy use of column metadata.
For implementing "factors" and other annotations, which is used by `scicloj.ml`.
So I think we need to think , if  a constructor taking additional "meta" is needed.

Otherwise asked: Should `tablecloth.column.api` become aware/support   the `tech.ml.dataset` ML related functions
in this three namespaces:
tech.v3.dataset.categorical
tech.v3.dataset.modelling
tech.v3.dataset.column-filters

Reply:
I am just thinking that it is not very important, that metadata can be given at column construction time.
In practice it will be added later to a column either by:
- inferring datatype during parsing of values
- use the 'modelling/factor' related function on the existing column

So the above `column` function does not need a way to specify metadata, I think now.

Reply:
@behrica thank you for the comments here. I think this is something that we can keep in mind going forward. This PR is just an initial step, not at all a final API. It can and will change before release. Having talked a bit more about logistics with @genmeblog, we will actually not put this PR into main. We are going to maintain a dev branch for the column API. So in summary let's keep this in mind. I will find a way of centralizing the conversation about this axis of the API in the next few days...